exp/query/api.go

```diff
@@ -79,21 +108,154 @@ func (a *API) Search(ctx context.Context, req *SearchRequest) (*SearchResponse,
 		return nil, err
 	}
 
-	// Expectations
-	// 0. Check if tenant status == FROZEN
-	// 1. Build local path `/<datapath>/<collection>/<tenant>` from search request
-	// 2. If it doesn't exist, fetch from object store to same path as above
-	// 3. If it doesn't exist on object store, return 404
-	// 4. Once in the local disk, load the index
-	// 5. Searve the search.
-	return &SearchResponse{}, nil
+	localPath := path.Join(a.offload.DataPath, strings.ToLower(req.Collection), strings.ToLower(req.Tenant), defaultLSMRoot)
+
+	// TODO(kavi): Avoid creating store every time?
+	store, err := lsmkv.New(localPath, localPath, a.log, nil, cyclemanager.NewCallbackGroupNoop(), cyclemanager.NewCallbackGroupNoop())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create store to read offloaded tenant data: %w", err)
+	}
+	defer store.Shutdown(ctx)
+
+	var resp SearchResponse
+
+	if len(req.NearText) != 0 {
+		// do vector search
```

Comment:
NIT: What do you think about splicing both search out in their own function and simply have `if len(req.NearText) != 0 { vectorSearch(...)} else { allObjects(...)}` ?